opacus/scheduler.py

```diff
@@ -0,0 +1,83 @@
+from typing import Callable
+
+from .optimizer import DPOptimizer
+
+
+class _NoiseScheduler(object):
+    def __init__(self, optimizer: DPOptimizer, last_epoch=-1):
+        # Attach optimizer
+        if not isinstance(optimizer, DPOptimizer):
+            raise TypeError("{} is not a DPOptimizer".format(type(optimizer).__name__))
```

Comment:
Do we want to do that on top of the type hint?

Reply:
Hmm but the hint is not enforced right?

Reply:
I don't really have strong opinion on this - I see an appeal of having a user-friendly message for a potentially common mistake too.

I personally wouldn't have added it: after all, we don't check and enforce other parameters. And if someone isn't using IDE or mypy to highlight the typing issues, they've probably brought this upon themselves.

Note, that if it's not for this message, the code will still fail with "optimizer has no attiribute noise_multiplier" couple lines later - so it's only a matter of a different message